platform/pulp/platform/models/content.py

```diff
@@ -0,0 +1,94 @@
+"""
+Content related Django models.
+"""
+import hashlib
+
+from django.contrib.contenttypes import fields
+from django.db import models
+
+
+from pulp.platform.models import Model, MasterModel, Notes
+from pulp.platform.models.storage import StoragePath
+
+
+class Content(MasterModel):
+    """
+    A piece of managed content.
+
+    :cvar natural_key_fields: Tuple of natural fields.  Must be: models.Field.
+    :type natural_key_fields: tuple
+
+    Fields:
+
+    :cvar type: The content type.
+    :type type: models.TextField
+
+    Relations:
+
+    :cvar notes: Arbitrary information stored with the content.
+    :type notes: fields.GenericRelation
+    """
+    natural_key_fields = ()
+
+    type = models.TextField(blank=False, default=None)
+
+    notes = fields.GenericRelation(Notes)
+
+    def natural_key(self):
```

Comment:
Maybe this should be a property. I'm not a huge fan of properties in general, because it seems to really just save a pair of ```()``` while obscuring the fact that it's a derived value. That said, I thought I'd point out the option. I'd be happy either way.